packages/mdc-tab-scroller/foundation.js

```diff
@@ -0,0 +1,378 @@
+/**
+  * @license
+  * Copyright 2018 Google Inc. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License")
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+*/
+
+import MDCFoundation from '@material/base/foundation';
+import {cssClasses, strings} from './constants';
+/* eslint-disable no-unused-vars */
+import {MDCTabScrollerAnimation, MDCTabScrollerHorizontalEdges, MDCTabScrollerAdapter} from './adapter';
+import MDCTabScrollerRTL from './rtl-scroller';
+/* eslint-enable no-unused-vars */
+import MDCTabScrollerRTLDefault from './rtl-default-scroller';
+import MDCTabScrollerRTLNegative from './rtl-negative-scroller';
+import MDCTabScrollerRTLReverse from './rtl-reverse-scroller';
+
+/**
+ * @extends {MDCFoundation<!MDCTabScrollerAdapter>}
+ * @final
+ */
+class MDCTabScrollerFoundation extends MDCFoundation {
+  /** @return enum {string} */
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  /** @return enum {string} */
+  static get strings() {
+    return strings;
+  }
+
+  /**
+   * @see MDCTabScrollerAdapter for typing information
+   * @return {!MDCTabScrollerAdapter}
+   */
+  static get defaultAdapter() {
+    return /** @type {!MDCTabScrollerAdapter} */ ({
+      addClass: () => {},
+      removeClass: () => {},
+      eventTargetMatchesSelector: () => {},
+      setScrollContentStyleProperty: () => {},
+      getScrollContentStyleValue: () => {},
+      setScrollAreaScrollLeft: () => {},
+      getScrollAreaScrollLeft: () => {},
+      getScrollContentOffsetWidth: () => {},
+      getScrollAreaOffsetWidth: () => {},
+      computeScrollAreaClientRect: () => {},
+      computeScrollContentClientRect: () => {},
+    });
+  }
+
+  /** @param {!MDCTabScrollerAdapter} adapter */
+  constructor(adapter) {
+    super(Object.assign(MDCTabScrollerFoundation.defaultAdapter, adapter));
+
+    /** @private {boolean} */
+    this.shouldHandleInteraction_ = false;
```

Comment:
It might be worth documenting what purpose this serves, since I see it being checked not only in handleInteraction but also handleTransitionEnd.